src/Collection/CollectionTrait.php

```diff
@@ -135,10 +135,20 @@ public function map(callable $c)
      * {@inheritDoc}
      *
      */
-    public function reduce(callable $c, $zero)
+    public function reduce(callable $c, $zero = null)
     {
+        $skip = false;
+        if ($zero === null) {
+            $zero = $this->first();
+            $skip = true;
+        }
+
         $result = $zero;
         foreach ($this->_unwrap() as $k => $value) {
+            if ($skip) {
+                $skip = false;
+                continue;
```

Comment:
@markstory I remove the call to `first` and instead I'm now just grabbing the value in the first iteration and `continue`.